metalearners/metalearner.py

```diff
@@ -133,6 +135,41 @@ def _validate_n_folds_synchronize(n_folds: dict[str, int]) -> None:
         raise ValueError("Need at least two folds to use synchronization.")
 
 
+def _evaluate_model(
+    cfes: Sequence[CrossFitEstimator],
+    X: Sequence[Matrix],
+    y: Sequence[Vector],
+    scorers: Sequence[str | Callable],
+    model_kind: str,
+    is_oos: bool,
+    is_treatment: bool,
+    oos_method: OosMethod = OVERALL,
+) -> dict[str, float]:
+    """Helper function to evaluate all the models of the same model kind."""
+    prefix = f"{model_kind}_"
+    evaluation_metrics: dict[str, float] = {}
+    for idx, scorer in enumerate(scorers):
+        if isinstance(scorer, str):
+            scorer_str = scorer
+            scorer_call: Callable = get_scorer(scorer)
+        else:
+            scorer_str = f"custom_scorer_{idx}"
+            scorer_call = scorer
+        for i, cfe in enumerate(cfes):
+            if is_treatment:
+                treatment_variant = i + 1
+                index_str = f"{treatment_variant}_vs_0_"
+            else:
+                if len(cfes) == 1:
+                    index_str = ""
+                else:
+                    index_str = f"{i}_"
+            name = f"{prefix}{index_str}{scorer_str}"
```

Comment:
Did you consciously decide against nesting the dictionary and for concatenating strings?

Reply:
I preferred to return a plain dictionary as then it would be easier to convert it to a `dataframe` if needed (e.g. in #9 ). I would be open to changing it but I also think it would carry unnecessary nesting, for example in the case where we only have one model type, we would have multiple levels but the first one would just have one key.

Reply:
I see, sgtm!